### PR TITLE
VideoCommon: Support shader logic ops on Metal (Apple GPUs) and OpenGL ES

### PR DESCRIPTION
**Pull request by OatmealDome**
This PR adds a shader logic ops implementation for Metal on Apple Silicon GPUs and OpenGL ES on mobile GPUs using framebuffer fetch.

Fixes Kirby's Air Ride (https://bugs.dolphin-emu.org/issues/11843).

## Metal

All Apple Silicon GPUs support framebuffer fetch, so they will all gain logic ops support. However, Intel, AMD, and NVIDIA GPUs do not support it, so Intel Macs are not affected by this PR.

MoltenVK has been patched to enable SPIRV-Cross to convert Vulkan's subpass inputs into framebuffer fetch.

## OpenGL ES

Devices that support either ``GL_EXT_shader_framebuffer_fetch`` or ``GL_ARM_shader_framebuffer_fetch`` can take advantage of this PR.

## TODO

The following issues need to be resolved / investigated:

- [x] Dim output on Apple M1 GPU
- [x] ~~Verify Mario Kart Wii box isn't back~~ box appears on dev, not related to this PR
- [x] Test Android to make sure it isn't broken
- [x] #10214 is merged

### CODE REVIEW

File: Source/Core/VideoCommon/GXPipelineTypes.h [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace VideoCommon
 // As pipelines encompass both shader UIDs and render states, changes to either of these should
 // also increment the pipeline UID version. Incrementing the UID version will cause all UID
 // caches to be invalidated.
-constexpr u32 GX_PIPELINE_UID_VERSION = 3;  // Last changed in PR 9532
+constexpr u32 GX_PIPELINE_UID_VERSION = 4;  // Last changed in PR XXXX
```
Comment (phire) [inline Source/Core/VideoCommon/GXPipelineTypes.h:22]: PR 10215

**Comment by JMC47** [comment]
I checked the notes and was going to yell at people to test this, but it turns out I was the person that was supposed to test this.  Oops.  Verified as working.

BEFORE:  

KirbyBroken

AFTER:

KirbyAirRideFixed

**Review by JMC47 — APPROVED** [review]
Everything is working in other games to the degree you'd expect as well.

**Comment by OatmealDome** [comment]
@dolphin-emu-bot rebuild

**Comment by pizuz** [comment]
Are you planning to add the SPIRV-Cross patch to upstream?

**Comment by OatmealDome** [comment]
@pizuz No plans to. The patch is honestly a bit of a kludge. That sort of behaviour may not be wanted by other applications using MoltenVK.

**Pull request merged**